### DISCUSSION TITLE
Getting an error everytime I run a papis command

### DISCUSSION THREAD

**Discussion by Archie-2021** [papis/papis#867] [Q&A] [answered]
Like this:

```
$  papis --version
Error while loading entrypoint 'EntryPoint(name='isbn', value='papis.isbn:Importer', group='papis.importer')': No module named 'pkg_resources'.
Error while loading entrypoint 'EntryPoint(name='isbn', value='papis.isbn:explorer', group='papis.explorer')': No module named 'pkg_resources'.
papis, version 0.13
```
Also I noticed I don't have some of the commands mentioned in documentation (e.g. `papis cache`). I installed papis recently via `pip`. Is something wrong with my install?

**Comment by alexfikl** [comment]
Can you try installing `setuptools`? I don't think that was an explicit dependency in 0.13, but it may be needed for the entrypoints to work.

I'm mostly guessing, but `pkg_resources` is part of the `setuptools` package, so that might be it..

**Reply by Archie-2021** [reply]
> Can you try installing `setuptools`? I don't think that was an explicit dependency in 0.13, but it may be needed for the entrypoints to work.
> 
> I'm mostly guessing, but `pkg_resources` is part of the `setuptools` package, so that might be it..

Thanks, Is this a library or a binary file? The thing is that I use linux and on linux installing via pip can cause conflict so I used `pipx` command to install in a python isolated environment. Not sure how I should install additional packages to it.

**Reply by alexfikl** [reply, marked as answer]
Hmm, I'm on Linux too, but I don't use `pipx` unfortunately. From their documentation at https://github.com/pypa/pipx?tab=readme-ov-file#inject-a-package, this should work?
```
pipx inject papis setuptools
```

**Reply by Archie-2021** [reply]
> Hmm, I'm on Linux too, but I don't use `pipx` unfortunately. From their documentation at https://github.com/pypa/pipx?tab=readme-ov-file#inject-a-package, this should work?
> 
> ```
> pipx inject papis setuptools
> ```

Yeah that removed the error. Thank you!

**Reply by ghost** [reply]
@alexfikl I encountered this error as well when installing papis in a venv. Not sure if an error is expected in that case, to me it wasn't. Installing setuptools resolved the error.

Should setuptools not be included in this project's dependencies to resolve this for others as well? I'm not familiar with python packaging, but thought I'd suggest it.

**Reply by alexfikl** [reply]
Hi! We don't use `setuptools` directly anywhere in the code, so it shouldn't be a dependency. The plugins are handled by [stevedore](https://docs.openstack.org/stevedore/latest/) at the moment, so maybe it's not declaring a dependency if it needs it. What version of `stevedore` do you have? Can check with
```
python -c 'from importlib import metadata; print(metadata.version("stevedore"))'
```

I'll try to see if removing setuptools on my system gives the same errors..

**Reply by alexfikl** [reply]
Hm, it seems to work (at least on the latest git development version)..
```
>> python -c 'import setuptools'
Traceback (most recent call last):
File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'setuptools'
>> papis --version
papis, version 0.13
```

**Reply by ghost** [reply]
> What version of stevedore do you have?

In my venv 5.3.0

**Reply by alexfikl** [reply]
Hm.. a bit vexing, haha. Are you also just getting an error for the `isbn` plugin or for others as well?

**Reply by ghost** [reply]
For others as well

**Reply by alexfikl** [reply]
Can you detail your installation steps a bit? I haven't really managed to get any errors like that locally (at least with the latest git version) :\
